examples/java/spring-boot/src/main/java/uk/gov/api/springboot/infrastructure/fetcher/v1alpha/V1AlphaFetcher.java

```diff
@@ -0,0 +1,142 @@
+package uk.gov.api.springboot.infrastructure.fetcher.v1alpha;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SynchronousSink;
+import uk.gov.api.springboot.domain.model.Api;
+import uk.gov.api.springboot.domain.services.fetcher.Fetcher;
+import uk.gov.api.springboot.infrastructure.config.MdcFacade;
+import uk.gov.api.springboot.infrastructure.mappers.V1AlphaMapper;
+import uk.gov.api.springboot.infrastructure.models.metadata.v1alpha.BulkMetadataResponse;
+import uk.gov.api.springboot.infrastructure.models.metadata.v1alpha.ErrorResponse;
+
+public class V1AlphaFetcher implements Fetcher {
+  private static final Logger LOGGER = LoggerFactory.getLogger(V1AlphaFetcher.class);
+  private static final String CORRELATION_ID_HEADER = "correlation-id";
+
+  private final WebClient webClient;
+  private final V1AlphaMapper mapper;
+  private final MdcFacade mdc;
+
+  public V1AlphaFetcher(WebClient webClient, V1AlphaMapper mapper, MdcFacade mdc) {
+    this.webClient = webClient;
+    this.mapper = mapper;
+    this.mdc = mdc;
+  }
+
+  @Override
+  public List<Api> fetch(String baseUrl) {
+    String url = baseUrl + "/apis";
+    String correlationId = mdc.get(MdcFacade.CORRELATION_ID);
+    mdc.put("request.v1alpha.correlation-id", correlationId);
+
+    ResponseEntity<BulkMetadataResponse> entity =
+        webClient
+            .get()
+            .uri(url)
+            .accept(MediaType.valueOf("application/vnd.uk.gov.api.v1alpha+json"))
+            .header(CORRELATION_ID_HEADER, correlationId)
+            .retrieve()
+            .onStatus(eq(HttpStatus.NOT_FOUND), handleNotFound(url))
```

Comment:
did we want this in the contract? It feels a bit weird saying it. Maybe one more for the specification for "how fetchers should work"

Reply:
having it in the spec instead makes sense to me